
# Project: Investigating Movie Data

## Table of Contents
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#wrangling">Data Wrangling</a></li>
<li><a href="#eda">Exploratory Data Analysis</a></li>
<li><a href="#conclusions">Conclusions</a></li>
<li><a href="#limitations">Limitations</a></li>
</ul>

<a id='intro'></a>
## Introduction

> The dataset is from The Movie Database (TMDb) which is a collection of information from about 10,000 movies. Information includes, title, cast, director, user ratings, revenue, budget, genre, release date, etc. The investigation will explore revenue and how they are affected by genres, popularity, ratings, etc. 

In [1]:
# importing libraries
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
%matplotlib inline

<a id='wrangling'></a>
## Data Wrangling


### General Properties

In [2]:
#read-in the dataset
df_movie = pd.read_csv('tmdb_movies.csv')
df_movie.head()

id    imdb_id  popularity     budget     revenue  \
0  135397  tt0369610   32.985763  150000000  1513528810   
1   76341  tt1392190   28.419936  150000000   378436354   
2  262500  tt2908446   13.112507  110000000   295238201   
3  140607  tt2488496   11.173104  200000000  2068178225   
4  168259  tt2820852    9.335014  190000000  1506249360   

                 original_title  \
0                Jurassic World   
1            Mad Max: Fury Road   
2                     Insurgent   
3  Star Wars: The Force Awakens   
4                     Furious 7   

                                                cast  \
0  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
1  Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
2  Shailene Woodley|Theo James|Kate Winslet|Ansel...   
3  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
4  Vin Diesel|Paul Walker|Jason Statham|Michelle ...   

                                            homepage          director  \
0                      http://www.jurassicworld.com/   Colin Trevorrow   
1                        http://www.madmaxmovie.com/     George Miller   
2     http://www.thedivergentseries.movie/#insurgent  Robert Schwentke   
3  http://www.starwars.com/films/star-wars-episod...       J.J. Abrams   
4                           http://www.furious7.com/         James Wan   

                         tagline  ...  \
0              The park is open.  ...   
1             What a Lovely Day.  ...   
2     One Choice Can Destroy You  ...   
3  Every generation has a story.  ...   
4            Vengeance Hits Home  ...   

                                            overview runtime  \
0  Twenty-two years after the events of Jurassic ...     124   
1  An apocalyptic story set in the furthest reach...     120   
2  Beatrice Prior must confront her inner demons ...     119   
3  Thirty years after defeating the Galactic Empi...     136   
4  Deckard Shaw seeks revenge against Dominic Tor...     137   

                                      genres  \
0  Action|Adventure|Science Fiction|Thriller   
1  Action|Adventure|Science Fiction|Thriller   
2         Adventure|Science Fiction|Thriller   
3   Action|Adventure|Science Fiction|Fantasy   
4                      Action|Crime|Thriller   

                                production_companies release_date vote_count  \
0  Universal Studios|Amblin Entertainment|Legenda...       6/9/15       5562   
1  Village Roadshow Pictures|Kennedy Miller Produ...      5/13/15       6185   
2  Summit Entertainment|Mandeville Films|Red Wago...      3/18/15       2480   
3          Lucasfilm|Truenorth Productions|Bad Robot     12/15/15       5292   
4  Universal Pictures|Original Film|Media Rights ...       4/1/15       2947   

   vote_average  release_year   budget_adj   revenue_adj  
0           6.5          2015  137999939.3  1.392446e+09  
1           7.1          2015  137999939.3  3.481613e+08  
2           6.3          2015  101199955.5  2.716190e+08  
3           7.5          2015  183999919.0  1.902723e+09  
4           7.3          2015  174799923.1  1.385749e+09  

[5 rows x 21 columns]

In [3]:
# identify datatypes for variables
df_movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10866 entries, 0 to 10865
Data columns (total 21 columns):
id                      10866 non-null int64
imdb_id                 10856 non-null object
popularity              10866 non-null float64
budget                  10866 non-null int64
revenue                 10866 non-null int64
original_title          10866 non-null object
cast                    10790 non-null object
homepage                2936 non-null object
director                10822 non-null object
tagline                 8042 non-null object
keywords                9373 non-null object
overview                10862 non-null object
runtime                 10866 non-null int64
genres                  10843 non-null object
production_companies    9836 non-null object
release_date            10866 non-null object
vote_count              10866 non-null int64
vote_average            10866 non-null float64
release_year            10866 non-null int64
budget_adj              1

* This shows the datatypes and counts of entries per column.
* There is a significant lack of data in the "homepage," "tagline," "keywords," and "production_companies," which those columns could probably be tossed out. 

In [4]:
# stats for numeric data
df_movie.describe()

id    popularity        budget       revenue       runtime  \
count   10866.000000  10866.000000  1.086600e+04  1.086600e+04  10866.000000   
mean    66064.177434      0.646441  1.462570e+07  3.982332e+07    102.070863   
std     92130.136561      1.000185  3.091321e+07  1.170035e+08     31.381405   
min         5.000000      0.000065  0.000000e+00  0.000000e+00      0.000000   
25%     10596.250000      0.207583  0.000000e+00  0.000000e+00     90.000000   
50%     20669.000000      0.383856  0.000000e+00  0.000000e+00     99.000000   
75%     75610.000000      0.713817  1.500000e+07  2.400000e+07    111.000000   
max    417859.000000     32.985763  4.250000e+08  2.781506e+09    900.000000   

         vote_count  vote_average  release_year    budget_adj   revenue_adj  
count  10866.000000  10866.000000  10866.000000  1.086600e+04  1.086600e+04  
mean     217.389748      5.974922   2001.322658  1.755104e+07  5.136436e+07  
std      575.619058      0.935142     12.812941  3.430616e+07  1.446325e+08  
min       10.000000      1.500000   1960.000000  0.000000e+00  0.000000e+00  
25%       17.000000      5.400000   1995.000000  0.000000e+00  0.000000e+00  
50%       38.000000      6.000000   2006.000000  0.000000e+00  0.000000e+00  
75%      145.750000      6.600000   2011.000000  2.085325e+07  3.369710e+07  
max     9767.000000      9.200000   2015.000000  4.250000e+08  2.827124e+09

> * So initially we see the counts for each of the numerical data has the same which is 10866 so there isn't any missing data for those columns. 
* The minimum for budget, revenue, budget_adj, and revenue_adj is 0 which would probably indicate the data maybe was not available considering I don't think a movie could be made without funds or would not have garnered any revenue. These will be investigated.
* The minimum for runtime is 0 as well which probably indicates an issue in data because movies have to have a runtime to exist. 
* The vote_average is logical because the scale is from 1-10 and the minimum and maximum are within those ranges with the mean being 5.975922 which is mostly centered. 
* The earliest release year is 1960 and latest is 2015 which is around when the dataset was created so that follows logic, with the mean release year being 2001. 
* The popularity rating seems skewed because the maximum is 32.985763 but the mean is 0.646441, so this will need further investigation. 

In [5]:
#query to fing movies with 0 minute runtimes
df_movie.query('runtime == 0')['original_title']

92           Mythica: The Necromancer
334                           Ronaldo
410                    Anarchy Parlor
445     The Exorcism of Molly Hartley
486                If There Be Thorns
595                         Deep Dark
616                      The Outfield
1241                        Dance-Off
1289                        Treehouse
1293                         Tim Maia
1849                     Spectacular!
2315             Listen to Your Heart
2370                  Freshman Father
3329         Grande, grosso e Verdone
3794             Toi, moi, les autres
3857                         Cell 213
3884                           eCupid
4063           Madea's Family Reunion
4138               A Time for Dancing
4829                             Rags
4944              How to Fall in Love
5216            Madea's Class Reunion
5695                 Skinwalker Ranch
5920           The Food Guide to Love
5938                      Go Goa Gone
5992                 Amiche da morire
6040        

> * Some of the titles are recognizable and because this is a userbased website where users contribute the data for these films, that is most likely the reason for 0 input for some of these titles. I could google search each runtime and enter it in manually but since it is such a small portion I'm going to just eliminate these rows most likely. 

In [6]:
#query for $0 budget movies
df_movie.query('budget_adj == 0')['original_title'].count()

5696

* A query for budget_adj equaling 0 was ran in order to see how many movies lacked budget info. Since this is a user-contributed site, that is most likely the reason for the lack of data and there are 5696 missing rows which would be impossible to fill manually. Also since a movie has to be funded somehow there is no point in analyzing a "free" movie. 



### Data Cleaning 

In [7]:
#dropped unnecessary columns
df_movie.drop(['homepage','imdb_id','tagline','keywords','overview', 'production_companies',
               'release_date', 'budget','revenue','cast', 'id'], axis=1, inplace=True)

In [8]:
df_movie.head()

popularity                original_title          director  runtime  \
0   32.985763                Jurassic World   Colin Trevorrow      124   
1   28.419936            Mad Max: Fury Road     George Miller      120   
2   13.112507                     Insurgent  Robert Schwentke      119   
3   11.173104  Star Wars: The Force Awakens       J.J. Abrams      136   
4    9.335014                     Furious 7         James Wan      137   

                                      genres  vote_count  vote_average  \
0  Action|Adventure|Science Fiction|Thriller        5562           6.5   
1  Action|Adventure|Science Fiction|Thriller        6185           7.1   
2         Adventure|Science Fiction|Thriller        2480           6.3   
3   Action|Adventure|Science Fiction|Fantasy        5292           7.5   
4                      Action|Crime|Thriller        2947           7.3   

   release_year   budget_adj   revenue_adj  
0          2015  137999939.3  1.392446e+09  
1          2015  137999939.3  3.481613e+08  
2          2015  101199955.5  2.716190e+08  
3          2015  183999919.0  1.902723e+09  
4          2015  174799923.1  1.385749e+09

* Got rid of the following columns: homepage, imdb_id, tagline, keywords, overview, production_companies, release_date, budget, revenue, cast
* These lacked data in most rows and didnt seem relevant to the information trying to be obtained.
* Also we have the release year so release date was tossed out and the adjusted revenue and budget based on inflation is included so the original values were not deemed useful for analysis.

In [ ]:
#query rows with $0 for budget and revenue
df_movie3 = df_movie2.query('revenue_adj != 0').query('budget_adj != 0')
df_movie3.info()

* removed rows where adjusted budget and revenue was 0 because it was missing data which would skew the results

In [ ]:
#cleaning data
df_movie3.to_csv('movies_clean.csv', index=False)

In [ ]:
#split genres for movies
df_movie3['genres_2'] = df_movie['genres'].str.split('|').str[0]

* Wanted to drop the multiple genres for films and choose the first one as the official genre for the movie, so a new column 'genres_2' was created

In [ ]:
df_movie3.drop(['genres'], axis=1, inplace=True)

In [ ]:
df_movie3.drop(['id'], axis=1, inplace=True)

* Dropped the original genres and id columns because they were unnecessary for the analysis

In [ ]:
df_movie3.genres_2.value_counts()

* a more concise and workable categories for genres found and since there was only one type found under TV Movie, the row will be dropped

In [ ]:
df_movie3.query("genres_2 == 'TV Movie'")

In [ ]:
df_movie3.drop(8615, inplace=True)

In [ ]:
df_movie3.vote_average.describe(), df_movie3.popularity.describe()

In [ ]:
be_votes = [2.2, 5.7, 6.2, 6.7, 8.4]
be_pop = [0.001117, 0.463068, 0.797723, 1.368403, 32.985763]

In [ ]:
bin_names_votes = ['Low', 'Below Average', 'Above Average', 'High']
bin_names_pop = ['Non Popular', 'Semi Popular', 'Popular', 'Very Popular']

In [ ]:
df_movie3['rating_level'] = pd.cut(df_movie3['vote_average'], be_votes, labels=bin_names_votes)
df_movie3['pop_level'] = pd.cut(df_movie3['popularity'], be_pop, labels=bin_names_pop)

In [ ]:
df_movie3.info()

In [ ]:
df_movie3["pop_level"] = df_movie3["pop_level"].astype('object')
df_movie3["rating_level"] = df_movie3["rating_level"].astype('object')

* Changed new columns from type category to objects for easier use with labels on graphs for later. 

In [ ]:
df_movie3.info()

In [ ]:
df_movie3.isnull().sum()

In [ ]:
df_movie3.dropna(axis=0, inplace=True)

In [ ]:
df_movie3.info()

* Found one NaN value in director column, one in the rating_level column, and one in the pop_level column. So these were removed and 3851 samples remain. 

In [ ]:
df_movie3.to_csv('movies_clean2.csv', index=False)

<a id='eda'></a>
## Exploratory Data Analysis



### Is the popularity of the movie associated with higher revenues?

In [ ]:
sns.relplot(x="popularity", y="revenue_adj", data=df_movie3);

* A simple scatter plot can show a positive correlation between level of popularity and revenue.
* A more appealing approach would be to categorize the levels of popularity based on the quartile levels of popularity. 
* Then a pie chart could visually see the variance in total revenue based on popularity. 

In [ ]:
tot_rev = df_movie3.groupby('pop_level')['revenue_adj'].sum()
tot_rev

In [ ]:
level_pops = df_movie3.pop_level.unique()
level_pops.sort()

In [ ]:
def pie_chart():
    sns.set(context="notebook")
    plt.pie(tot_rev, labels=level_pops, autopct='%1.1f%%')
    plt.legend(title="Popularity",
              loc="center right",
              bbox_to_anchor=(1.5, 0, 0.5, 1));

### Which genres of movies have the higher revenues on average?

In [ ]:
df_movie3.groupby('genres_2').revenue_adj.describe()


* the describe function can give general statistics on each genre of film and in reference to revenue
* Some boxplots of this data would be easier to see which genres fair better in terms of revenue. 

In [ ]:
boxplot_data = df_movie3.query('revenue_adj < 500000000')

* A query for revenue less than $500,000,000 was chosen to show where most of the data is clustered to make the boxplots more visually appealing. 

In [ ]:
def boxplot():
    sns.set(style="ticks", font_scale=1.75)
    sns.catplot(x="genres_2", y="revenue_adj", kind="box", height=10, aspect=2, data=boxplot_data)
    plt.xlabel("Genres")
    plt.ylabel("Revenue in $100,000,000s")
    plt.xticks(rotation=45);

In [ ]:
pie_chart()

In [ ]:
boxplot()

<a id='conclusions'></a>
## Conclusions


> **Pie Chart**: 
* The total revenue divided among the popularity levels shows that more popular movies generate higher revenues. This is most likely because the popularity of a movie generates more "hype" which can create interest with more people willing to see a movie. 
* According to the TMDb website, the popularity rating is based on:
     Number of votes for the day
     Number of views for the day
     Number of users who marked it as a "favourite" for the day
     Number of users who added it to their "watchlist" for the day
     Release date
     Number of total votes
     Previous days score
* Therefore any correlation between revenue and popularity is independent of one another and does not directly contribute to the score. 

> **Box Plot**:
* The boxplot shows that Animation has the overall highest sales of revenue between all the genres. 
* The median is significantly higher than the others which would indicate more data points in higher grossing revenue range. 
* Documentary had the lowest revenues along with Mystery.


<a id='limitations'></a>
## Limitations

* There was a substantial amount of movie data excluded based on missing data values in the budget and revenue columns. 
* Certain movies had multiple genres listed under them but only the first one listed was used in this analysis. 
* The sample size for certain genres was limited which might affect the results. 
